src/test/java/com/airlines/dao/impl/CrewDaoImplTest.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    @Test
+    public void shouldCorrectlyAddCrewMemberToCrew() throws SQLException {
+        populateTestData();
+        crewDao.add(CREW_MEMBER, CREW);
+        List<CrewMember> list = crewDao.getByCrewId(CREW.getId());
+        assertTrue(list.contains(CREW_MEMBER));
+    }
+
+    @Test
+    public void shouldGetCrewMembersByCrewId() throws SQLException {
+        populateTestData();
+        List<CrewMember> list = crewDao.getByCrewId(1L);
+        assertEquals(3, list.size());
+
+        CrewMember crewMember1 = list.get(0);
+        CrewMember crewMember2 = list.get(1);
+        CrewMember crewMember3 = list.get(2);
+
+        assertEquals("Igor", crewMember1.getFirstName());
+        assertEquals("Shpack", crewMember1.getLastName());
+
+        assertEquals("Igor", crewMember2.getFirstName());
+        assertEquals("Tkachenko", crewMember2.getLastName());
+
+        assertEquals("Egor", crewMember3.getFirstName());
+        assertEquals("Shleeman", crewMember3.getLastName());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfCrewIsNotProvidedForGetByIdMethod() {
+        Crew invalidCrew = buildInvalidCrewFromDataSet();
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.getByCrewId(invalidCrew.getId()));
+        assertEquals("Cannot find crew members for crew without id", exception.getMessage());
+    }
+
+    @Test
+    public void getByCrewName() throws SQLException {
+        populateTestData();
+        List<CrewMember> list = crewDao.getByCrewName("Vityaz");
+        assertEquals(3, list.size());
+
+        CrewMember crewMember1 = list.get(0);
+        CrewMember crewMember2 = list.get(1);
+        CrewMember crewMember3 = list.get(2);
+
+        assertEquals("Igor", crewMember1.getFirstName());
+        assertEquals("Shpack", crewMember1.getLastName());
+
+        assertEquals("Igor", crewMember2.getFirstName());
+        assertEquals("Tkachenko", crewMember2.getLastName());
+
+        assertEquals("Egor", crewMember3.getFirstName());
+        assertEquals("Shleeman", crewMember3.getLastName());
+    }
+
+
+    @Test
+    public void shouldRemoveCrewMemberFromCrew() throws SQLException {
+        populateTestData();
+
+        crewDao.removeCrewMemberFromCrew(CREW, CREW_MEMBER);
+
+        List<CrewMember> listAfterRemove = crewDao.getByCrewId(1L);
+        assertEquals(2, listAfterRemove.size());
+        assertFalse(listAfterRemove.contains(CREW_MEMBER));
+    }
+
+    @Test
+    public void shouldThrowExceptionIfCrewMemberWithNoIdProvidedToRemoveMethod() {
+        CrewMember crewMember = CrewMember.builder()
+                .withId(null)
+                .build();
+
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.removeCrewMemberFromCrew(CREW, crewMember));
+
+        assertEquals("Valid crewMember entity should be provided", exception.getMessage());
+    }
+
```

Comment:
same as for add method, in first commit you had 2 tests, you should write 2 tests more to cover 4 cases.
4 cases == 4 tests, as it seems to me, you just remove your 2 tests and add 2 other tests, so nothing changed - from 2 tests to 2 tests.

But it should be 4 tests, because there we have two IF and each IF contain || - two conditions it covers
2 conditions * 2 IF = 4 possible scenarios.
Each of them should be tested